### PR TITLE
Write support for DWRF encryption at rest

### PR DESCRIPTION
**Pull request by mayankgarg1990**
Flow `EncryptionInformation` to `HiveOutputTableHandle` and `HiveInsertTableHandle` through `HiveWritableTableHandle`. Extended the `HiveEncryptionInformationProvider` to also provide the `EncryptionInformation` necessary for when we are writing data.

In addition to this, did some refactor in `HiveMetadata` to better re-use code for translating settings between table properties and hive properties for both read and write cases.

```
== NO RELEASE NOTE ==
```

Depended by https://github.com/facebookexternal/presto-facebook/pull/1053

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMetadataFileFormatEncryptionSettings.java [hunk 1]
```diff
@@ -125,13 +143,15 @@ public void tearDown()
         }
     }
 
-    private static ConnectorTableMetadata getConnectorTableMetadata(String tableName, Map<String, Object> tableProperties)
+    private static ConnectorTableMetadata getConnectorTableMetadata(String tableName, Map<String, Object> tableProperties, boolean isPartitioned)
```
Comment (rschlussel) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMetadataFileFormatEncryptionSettings.java:146]: Maybe I missed it, but I think none of these tests are using unpartitioned tables. Maybe add a test for that?

Comment (mayankgarg1990) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveMetadataFileFormatEncryptionSettings.java:146]: There were already `CTAS` tests to which I amended the write side of the tests. That said, you are correct that I need to add tests for `CREATE` and `INSERT INTO` - let me see if the code supports that or not :)

**Comment by rschlussel** [comment]
Also, I didn't follow how is the information being routed back to the TableFinishOperator? I expected it would need to go inside  the PartitionUpdate, but it doesn't look like you're doing that.

**Comment by mayankgarg1990** [comment]
> Also, I didn't follow how is the information being routed back to the TableFinishOperator? I expected it would need to go inside the PartitionUpdate, but it doesn't look like you're doing that.

This happens via the TableHandle object which as per my understanding is the same as we generate during `beginInsert`

**Pull request merged**